src/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.Unix.cs

```diff
@@ -205,9 +205,43 @@ private void InnerReleaseHandle()
 
         internal sealed partial class InnerSafeCloseSocket : SafeHandleMinusOneIsInvalid
         {
+            private Interop.Error CloseHandle(IntPtr handle)
+            {
+                Interop.Error errorCode = Interop.Error.SUCCESS;
+                bool remappedError = false;
+
+                if (Interop.Sys.Close(handle) != 0)
+                {
+                    errorCode = Interop.Sys.GetLastError();
+                    if (errorCode == Interop.Error.ECONNRESET)
+                    {
+                        // Some Unix platforms (e.g. FreeBSD) non-compliantly return ECONNRESET from close().
+                        // For our purposes, we want to ignore such a "failure" and treat it as success.
+                        // In such a case, the file descriptor was still closed and there's no corrective
+                        // action to take.
+                        errorCode = Interop.Error.SUCCESS;
+                        remappedError = true;
+                    }
+                }
+
+                if (NetEventSource.IsEnabled)
```

Comment:
Does this need an "if the handle is being closed for any reason other than AppDomain shutdown" check surrounding it?

Reply:
Would it make sense to raise exception or return error code? My concern is that checking return code on Close() is inly one way how to know your data has been delivered. Since write()/send()may simply put it to socket buffer there is no way for application to know something did not work as expected if both write/send and close succeed. 
```
man 2 linux
   Dealing with error returns from close()
       A  careful programmer will check the return value of close(), since it is quite possible that errors on a previous write(2) operation are reported only on the final close() that releases the open file description.
       Failing to check the return value when closing a file may lead to silent loss of data.  This can especially be observed with NFS and with disk quota.
```
It seems like it can return EIO bit i don't know if it does in practice on Linux.

Reply:
>> Does this need an "if the handle is being closed for any reason other than AppDomain shutdown" check surrounding it?

> Why?

@GrabYourPitchforks, I'm going to go ahead and merge this, as this isn't introducing any logging that wasn't already there.  But let's follow-up on your concerns to see if there's something that needs to be addressed.